plugin.py

```diff
@@ -43,6 +44,18 @@ def get_additional_variables(cls) -> Optional[Dict[str, str]]:
     def minimum_node_version(cls) -> Tuple[int, int, int]:
         return (10, 0, 0)
 
+    @classmethod
+    def install_or_update(cls) -> None:
+        super().install_or_update()
+        settings, _ = cls.read_settings()
+        cache_path = settings.get("initializationOptions", {}).get("storagePath", "")
+        variables = sublime.active_window().extract_variables()
+        variables.update(cls.additional_variables())
+
+        cache_path = os.path.realpath(os.path.abspath(sublime.expand_variables(cache_path, variables)))
+        if cache_path and os.path.isdir(cache_path):
+            shutil.rmtree(cache_path, ignore_errors=True)
```

Comment:
You could shortcut the code and do nothing if the initial value of `cache_path` is falsy.

But also there are more fundamental issues with this logic:
 - This doesn't consider project overrides where the path might be different.
 - If the path is unset then I guess intelephense still saves the cache to the default location but then we won't clear it?
 - I think ideally we would just pass `clearCache` when appropriate but I think "appropriate" might be hard to determine currently. I would need to invest more time into this to be able to maybe figure out a better solution.

Reply:
Also this code seems very dangerous. If someone would accidentally set '~' or `/` as the path then I'm guessing bad things would happen.

Reply:
> Also this code seems very dangerous. If someone would accidentally set '~' or `/` as the path then I'm guessing bad things would happen.

We should add Intelephense after the user directory to ensure directory security

Reply:
> * I think ideally we would just pass `clearCache` when appropriate but I think "appropriate" might be hard to determine currently. I would need to invest more time into this to be able to maybe figure out a better solution.

^ this +1

Reply:
> > * I think ideally we would just pass `clearCache` when appropriate but I think "appropriate" might be hard to determine currently. I would need to invest more time into this to be able to maybe figure out a better solution.
> 
> ^ this +1

Okay, but the old cache files will always exist and grow

Reply:
Ah, wait, `clearCache` only clean the current project but not all project?

Reply:
> Ah, wait, `clearCache` only clean the current project but not all project?

Yes, The server will occasionally create a directory similar to the one in the screenshot

![image](https://user-images.githubusercontent.com/49985266/119106967-66845d00-ba51-11eb-8f32-4686c8f48beb.png)

`clearcache` only clears the files in the directory, it does not delete the directory, so the directory grows

Reply:
> > Ah, wait, `clearCache` only clean the current project but not all project?
> 
> Yes, The server will occasionally create a directory similar to the one in the screenshot
> 
> ![image](https://user-images.githubusercontent.com/49985266/119106967-66845d00-ba51-11eb-8f32-4686c8f48beb.png)
> 
> `clearcache` only clears(It's actually an update) the files in the directory, it does not delete the directory, so the directory grows

These two directories are the same project, and I guess that's why bmewburn/vscode-intelephense#1523  can't be provided